app/models/ext_management_system.rb

```diff
@@ -65,7 +65,10 @@ def self.create_from_params(params, endpoints, authentications)
     end
   end
 
-  belongs_to :provider
+  # Typically belongs_to wouldn't dependent destroy but currently all management of
+  # Providers is done via the Manager, so when we delete the Manager we have to
+  # dependent delete the provider.
+  belongs_to :provider, :dependent => :destroy
```

Comment:
Doesn't this also need the `autosave`? Or you want to do it in a separate PR?

Reply:
Yeah for the foreman changes we need that, but two other tests started failing with that so I decided to pull it out into its own PR